ngrinder-controller/src/main/java/org/ngrinder/perftest/service/PerfTestRunnable.java

```diff
@@ -391,6 +391,18 @@ void startAgentsOn(PerfTest perfTest, GrinderProperties grinderProperties, Singl
 	 * @param grinderProperties grinder properties
 	 * @param singleConsole     console to be used.
 	 */
+
+
+	private boolean ignoreStop(Config config, StopReason stopReason) {
+		return (stopReason == StopReason.TOO_LOW_TPS &&
+				config.getControllerProperties().getPropertyBoolean(PROP_CONTROLLER_IGNORE_TOO_LOW_TPS, false)) ||
+			   (stopReason == StopReason.TOO_MANY_ERRORS &&
+				config.getControllerProperties().getPropertyBoolean(PROP_CONTROLLER_IGNORE_TOO_MANY_ERRORS, false)) ||
+			   (stopReason == StopReason.TOO_MUCH_TRAFFIC_ON_REGION &&
+				config.getControllerProperties().getPropertyBoolean(PROP_CONTROLLER_IGNORE_TOO_MUCH_TRAFFIC_ON_REGION,
+					false));
```

Comment:
Too complex.
Please make multiple if statement.

like..

```
if (stopReason == StopReason.TOO_LOW_TPS &&  	config.getControllerProperties().getPropertyBoolean(PROP_CONTROLLER_IGNORE_TOO_LOW_TPS))  {
     return false;
}
```

Reply:
i thought that return with one line condition is more efficient than using multiple if statement.
but it's too long and complex like you pointed out!

As a junior developer, I have a question!

> 1. Why it's better assigning and using a variable than using a getter by calling it multiple times? it is because of efficiency or because of readability?
> 
> 2. Choosing multiple if statements instead of one long comparison expression comparing conditions does not make a big difference in performance, so it is more efficient to select more readable code, or to use equal or multiple if statements. I wonder if that's why they're using it.

Reply:
I think the performance difference is negligible here so improve readability more valuable.

Reply:
thanks!!